File: Python Resources/resources/ds/disjoint_set.py
class DisjointSet:
    def __init__(self, n):
        self.parent = [i for i in range(n)]
        self.size = [1] * n

    def find(self, i):
        if self.parent[i] == i:
            return i
        else:
            self.parent[i] = self.find(self.parent[i])
            return self.parent[i]

    def union(self, a, b):
        x, y = self.find(a), self.find(b)
        if x != y:
            self.size[y] += self.size[x]
            self.parent[x] = self.parent[b]

    def get_size(self, i):
        return self.size[self.find(i)]

File: Resources/Java/src/resources/ds/DisjointSet.java
package resources.ds;

/**
 * @author Andi Gu
 */
public class DisjointSet {
    private int[] parent; //parent
    private int[] size; //size of the set

    public DisjointSet(int n) {
        parent = new int[n];
        size = new int[n];
        for (int i = 0; i < n; i++) {
            parent[i] = i;
            size[i] = 1;
        }
    }

    private int find(int i) {
        if (parent[i] == i) {
            return i;
        }
        return parent[i] = find(parent[i]);
    }

    public void union(int a, int b) {
        if (find(a) != find(b)) {
            size[find(b)] += size[find(a)];
            parent[find(a)] = parent[b];
        }
    }

    public int getSize(int i) {
        return size[find(i)];
    }
}
